ckan/tests/lib/test_base.py

```diff
@@ -18,31 +20,45 @@ def test_comment_absent_if_debug_false(app):
     assert "<!-- Snippet " not in response
 
 
-def test_apikey_missing(app):
+def test_apitoken_missing(app):
     request_headers = {}
 
     app.get("/dataset/new", headers=request_headers, status=403)
 
 
 @pytest.mark.usefixtures("clean_db", "with_request_context")
-def test_apikey_in_authorization_header(app):
+def test_apitoken_in_authorization_header(app):
     user = factories.Sysadmin()
-    request_headers = {"Authorization": str(user["apikey"])}
+    user_token = h.call_action(
+        "api_token_create",
+        context={"model": model, "user": user["name"]},
+        user=user["name"],
+        name="first token",
+    )
+    request_headers = {
+        "Authorization": user_token["token"]
+    }
 
     app.get("/dataset/new", headers=request_headers)
 
 
 @pytest.mark.usefixtures("clean_db", "with_request_context")
-def test_apikey_in_x_ckan_header(app):
+def test_apitoken_in_x_ckan_header(app):
     user = factories.Sysadmin()
+    user_token = h.call_action(
```

Comment:
This pattern looks like it's going to be used a lot across the tests, so why don't we add a helper to make life easier for developers. Why don't we create a factory for them, eg `user_token = factories.APIToken(user=user_id)`

If you look at `factories.py` it should be easy to adapt an existing one calling the approapiate action.

Reply:
Yep, you are right, i will do it.
Thanks for your suggestion.